### PR TITLE
fix(queue): add message when branch protection is misconfigured

### PR DESCRIPTION
**Pull request by sileht**
batch_size and speculative_checks requires branch protection strict mode
to be disabled.

### CODE REVIEW

**Review by jd — COMMENTED** [review]
Can we add a test? 😁

**Comment by sileht** [comment]
> Can we add a test? 😁

mypy should have catches that..., but it's not implemented yet: https://github.com/python/mypy/issues/12094

File: mergify_engine/actions/queue.py [hunk 1]
```diff
@@ -228,6 +228,28 @@ async def run(
...
+        if (
+            protection
+            and "required_status_checks" in protection
+            and "strict" in protection["required_status_checks"]
+            and protection["required_status_checks"]["strict"]
+        ):
+            if self.queue_rule.config["batch_size"] > 1:
+                return check_api.Result(
+                    check_api.Conclusion.FAILURE,
+                    "batch_size > 1 is not compatible with branch protection a setting",
```
Comment (Syffe) [inline mergify_engine/actions/queue.py:243]:
```suggestion
                    "batch_size > 1 is not compatible with branch protection setting",
```

File: mergify_engine/actions/queue.py [hunk 2]
```diff
@@ -228,6 +228,28 @@ async def run(
...
+                    "The branch protection setting `Require branches to be up to date before merging` must be unset.",
+                )
+            elif self.queue_rule.config["speculative_checks"] > 1:
+                return check_api.Result(
+                    check_api.Conclusion.FAILURE,
+                    "speculative_checks > 1 is not compatible with branch protection a setting",
```
Comment (Syffe) [inline mergify_engine/actions/queue.py:249]:
```suggestion
                    "speculative_checks > 1 is not compatible with branch protection setting",
```

File: docs/source/actions/queue.rst [hunk 3]
```diff
@@ -213,6 +213,13 @@ For example, if your queue is 15 pull requests long and you have
 draft pull requests of 5 pull requests each being tested at the same time. If
 your CI time is 10 min, you can merge those 15 pull requests in only 10 minutes.
 
+.. warning::
+
+   With batch size, pull requests are tested against the base branch
+   within another temporary pull request. This requires the branch protection
+   settings ``Require branches to be up to date before merging`` to be
+   disabled. If you require a linear history, just set the queue option ``merge: rebase``.
```
Comment (jd) [inline docs/source/actions/queue.rst:221]:
```suggestion
   With batch mode, pull requests are tested against the base branch
   within another temporary pull request. This requires the branch protection
   settings ``Require branches to be up to date before merging`` to be
   disabled. If you require a linear history, just set the queue option ``merge: rebase``.
```

**Review by jd — CHANGES_REQUESTED** [review]

**Comment by jd** [comment]
@mergifyio squash

**Review by jd — APPROVED** [review]

**Pull request merged**